File: sample/src/main/java/me/xiaopan/sketchsample/bean/Image.java
package me.xiaopan.sketchsample.bean;

import android.os.Parcel;
import android.os.Parcelable;

public class Image implements Parcelable{
    public String regularUrl;
    public String highDefinitionUrl;

    public Image(String regularUrl, String highDefinitionUrl) {
        this.regularUrl = regularUrl;
        this.highDefinitionUrl = highDefinitionUrl;
    }

    protected Image(Parcel in) {
        regularUrl = in.readString();
        highDefinitionUrl = in.readString();
    }

    public static final Creator<Image> CREATOR = new Creator<Image>() {
        @Override
        public Image createFromParcel(Parcel in) {
            return new Image(in);
        }

        @Override
        public Image[] newArray(int size) {
            return new Image[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(regularUrl);
        dest.writeString(highDefinitionUrl);
    }
}

File: sample/src/main/java/me/xiaopan/sketchsample/bean/Image.kt
package me.xiaopan.sketchsample.bean

import android.os.Parcel
import android.os.Parcelable

class Image() : Parcelable {
    lateinit var normalQualityUrl: String
    lateinit var rawQualityUrl: String

    constructor(parcel: Parcel) : this() {
        normalQualityUrl = parcel.readString()
        rawQualityUrl = parcel.readString()
    }

    constructor(regularUrl: String, highDefinitionUrl: String) : this() {
        this.normalQualityUrl = regularUrl
        this.rawQualityUrl = highDefinitionUrl
    }

    override fun toString(): String {
        return String.format("Image{%s -> %s}", normalQualityUrl, rawQualityUrl)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(normalQualityUrl)
        parcel.writeString(rawQualityUrl)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Image> {
        override fun createFromParcel(parcel: Parcel): Image {
            return Image(parcel)
        }

        override fun newArray(size: Int): Array<Image?> {
            return arrayOfNulls(size)
        }
    }
}
